### ISSUE TITLE
Warn site admins that extensions have read/write access to site-admin graphql commands

### ISSUE THREAD

**Issue by francisschmaltz** [sourcegraph/extensions-client-common#62]
There should be a prompt for site-admins warning that extensions have read/write access the the graphql API endpoint including site-admin commands.

**Comment by slimsag** [comment]
To be clear, an extension can only do what a user could regularly do. If the user running the extension is an admin, it can perform admin actions. If the user running the extension is not an admin, it cannot.

I'll update the alert we have when adding extensions to communicate this:

![image](https://user-images.githubusercontent.com/3173176/47465428-d6e77680-d7a1-11e8-96ea-1b3b8417aa19.png)

**Comment by francisschmaltz** [comment]
We should make sure this communicates to admins that it can read/change Sourcegraph admin settings

**Comment by slimsag** [comment]
I do not think explicitly calling that out makes sense. It will just pollute that dialog.

Read/change admin settings is covered under what I said:

> If the user running the extension is an admin, it can perform admin actions.

**Comment by francisschmaltz** [comment]
I think it's worth calling out. Sourcegraph settings sounds like my user settings not site settings. We need to make this is clear to an admin before enabling.

**Comment by slimsag** [comment]
This is what I will change to:

> It can:
>
> - Read repositories and files you view using Sourcegraph.
> - Read and change your Sourcegraph settings.
> - If the user running the extension is an admin, it can perform the above for admin-only data.

**Comment by francisschmaltz** [comment]
Can it say "site-admin data" rather than "admin-only" ?

**Comment by slimsag** [comment]
Changed to "admin data" -- "site-admin data" makes it sound like it only applies to site-specific admin data, which is not true: it's all admin data

**Comment by francisschmaltz** [comment]
I get that it's all data but the point is to call out that it's getting the site admin data so it'd be helpful to call that out so there isn't any confusion

**Comment by slimsag** [comment]
I don't understand. Can you explain what you think the difference is between "site admin data" and "admin data"?

**Comment by francisschmaltz** [comment]
"Site admin" should specifically be used so that it is clear that the same data on the site admin page can be accessed by the extension, including license key, git host tokens, etc.

**Comment by slimsag** [comment]
So is this perhaps something that should be in our style-guide then? we should always call admin data "site admin data"?

**Comment by francisschmaltz** [comment]
Yeah, I can add it in there. We refer to the page as "Site admin" so we should refer to those items as "site admin data" to avoid confusion

**Comment by francisschmaltz** [comment]
This also will change soon with the management console – extensions and site admins won't have access to that.

**Comment by slimsag** [comment]
Can you elaborate on your last comment? It does not make sense to me.
